### DISCUSSION TITLE
Definition of "done"

### DISCUSSION THREAD

**Discussion by zacharytomlinson** [zacharytomlinson/saspy#9] [General]
There has been some discussion with @dedalgr about the nuances of certain machines and their behaviors with certain commands/events. My thoughts are we will never be able to make code that will satisfy all machines everywhere, and really we shouldn't need to. We are not aiming to make a "SAS executable" with long-lived maintained "states" with regard to the connection to SAS. We are just providing the utilities to allow an end user to write an executable using our library. It can be left to their responsibility to make the executable work as they need it for their use case, we just need to provide the generalized methods with sufficient abstraction to allow them to build their executable without extending saspy's functionality. Thus as we talk about different mechanisms of issuing certain commands in certain orders, and then calling a cleanup event poll function and all of that, we are trying to fix the executable that will be made by the end user and not focusing on just providing the utilities they need to build their own. So my thoughts would be as opposed to talking about the strategy behind issuing multiple different types of commands and in what order, we focus on just making saspy sufficiently generalized and abstract as to empower the end user to build a good executable with little to no modification/extension, as opposed to building a robust executable (for example in example.py). Your thoughts on this @well-it-wasnt-me ? Let me know if I need to clarify any more.

**Comment by well-it-wasnt-me** [comment]
All clear. And I agree. This is a library and not an end executable. 

Since having documentation about SAS Protocol is very hard (to obtain and to understand) what we can do is to provide a comprehensive WiKi and/or Doc where we tackle all those issues 
(Which command to send and which order, why there is the needing to do that and so on...) and make the library at the best of our capabilities.

**Reply by zacharytomlinson** [reply]
Love the idea!
